cmd/sqlcmd/sqlcmd.go

```diff
@@ -118,23 +120,170 @@ func (a SQLCmdArguments) authenticationMethod(hasPassword bool) string {
+	rootCmd.Flags().StringVarP(&args.EncryptConnection, "encrypt-connection", "N", "default", "This switch is used by the client to request an encrypted connection.")
+	rootCmd.Flags().StringVarP(&args.Format, "format", "F", "horiz", "Specifies the formatting for results.")
+	rootCmd.Flags().IntVarP(&args.ErrorsToStderr, "errors-to-stderr", "r", -1, "Controls which error messages are sent to stdout. Messages that have severity level greater than or equal to this level are sent.")
+
+	rootCmd.Flags().IntVar(&args.DriverLoggingLevel, "driver-logging-level", 0, "Level of mssql driver messages to print.")
+	rootCmd.Flags().BoolVarP(&args.ExitOnError, "exit-on-error", "b", false, "Specifies that sqlcmd exits and returns a DOS ERRORLEVEL value when an error occurs.")
+	rootCmd.Flags().IntVarP(&args.ErrorLevel, "error-level", "m", 0, "Controls which error messages are sent to stdout. Messages that have severity level greater than or equal to this level are sent.")
+
+	//Need to decide on short of Header , as "h" is already used in help command in Cobra
+	rootCmd.Flags().IntVarP(&args.Headers, "headers", "h", 0, "Specifies the number of rows to print between the column headings. Use -h-1 to specify that headers not be printed.")
 
-	// so far sqlcmd prints all the errors itself so ignore it
-	exitCode, _ := run(vars, &args)
-	os.Exit(exitCode)
+	rootCmd.Flags().BoolVarP(&args.UnicodeOutputFile, "unicode-output-file", "u", false, "Specifies that all output files are encoded with little-endian Unicode")
+	rootCmd.Flags().StringVarP(&args.ColumnSeparator, "column-separator", "s", "", "Specifies the column separator character. Sets the SQLCMDCOLSEP variable.")
+	rootCmd.Flags().BoolVarP(&args.TrimSpaces, "trim-spaces", "W", false, "Remove trailing spaces from a column.")
+	rootCmd.Flags().BoolVarP(&args.MultiSubnetFailover, "multi-subnet-failover", "M", false, "Provided for backward compatibility. Sqlcmd always optimizes detection of the active replica of a SQL Failover Cluster.")
+
+	rootCmd.Flags().StringVarP(&args.Password, "password", "P", "", "Obsolete. The initial passwords must be set using the SQLCMDPASSWORD environment variable or entered at the password prompt.")
+
+	// Using PersistentFlags() for ErrorSeverityLevel due to data type uint8 , which is not supported in Flags()
+	rootCmd.PersistentFlags().Uint8VarP(&args.ErrorSeverityLevel, "error-severity-level", "V", 0, "Controls the severity level that is used to set the ERRORLEVEL variable on exit.")
+	// ScreenWidth is *int type , The IntVarP() method takes a pointer to an integer variable as its first argument, which it binds to the flag.
+	args.ScreenWidth = new(int)
+	rootCmd.Flags().IntVarP(args.ScreenWidth, "screenwidth", "w", 0, "Specifies the screen width for output. Sets the SQLCMDCOLWIDTH variable.")
```

Comment:
>args.ScreenWidth [](http://example.com/codeflow?start=25&length=16)

is args.ScreenWidth left as nil when the flag is not passed?

Reply:
here's the test for this:
```
set SQLCMDCOLWIDTH=20
16:36:41.50 e:\git>sqlcmd 
1> :listvar
SQLCMDCOLSEP = " "
SQLCMDCOLWIDTH = "20"
```

Notice we preserve the environment variable if the command line flag is not used.

By contrast:
```
16:38:16.97 e:\git>sqlcmd -w 50
1> :listvar
SQLCMDCOLSEP = " "
SQLCMDCOLWIDTH = "50"
```